solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java

```diff
@@ -93,9 +95,10 @@
   // This executor is initialized in the init method
   private ExecutorService commExecutor;
 
+  protected volatile List<Http2SolrClient> multipleClients = new ArrayList<>(CLIENT_COUNT);
   protected volatile Http2SolrClient defaultClient;
```

Comment:
If we want to adopt the approach of allowing multiple clients, perhaps we can consider removing this field. As in the new design, this field can be null (if `CLIENT_COUNT` > 1), such is not that obvious for developer and one might accidentally use this field w/o the null check. The only place that still reference `defaultClient` is `getShardHandler`, perhaps we can just do 
```
public ShardHandler getShardHandler() {
    return getShardHandler(multipleClients.size == 1 ? multipleClients.get(0) :  multipleClients.get( r.nextInt(multipleClients.size())));
  }
```
 ?

Or if we still want to maintain a convenient handle for such `defaultClient`, add a getter method that uses the random logic to select the client? 😊

Reply:
I'm fine either way. It was just a micro optimization